platform/linux-generic/odp_ipsec.c

```diff
@@ -1521,17 +1521,13 @@ static ipsec_sa_t *ipsec_out_single(odp_packet_t pkt,
 	}
 
 	if (!crypto.ok) {
-		if ((crypto.cipher_status.alg_err !=
-		     ODP_CRYPTO_ALG_ERR_NONE) ||
-		    (crypto.cipher_status.hw_err !=
-		     ODP_CRYPTO_HW_ERR_NONE))
-			status->error.alg = 1;
-
-		if ((crypto.auth_status.alg_err !=
-		     ODP_CRYPTO_ALG_ERR_NONE) ||
-		    (crypto.auth_status.hw_err !=
-		     ODP_CRYPTO_HW_ERR_NONE))
+		if ((crypto.cipher_status.alg_err ==
+		     ODP_CRYPTO_ALG_ERR_ICV_CHECK) ||
+		    (crypto.auth_status.alg_err ==
+		     ODP_CRYPTO_ALG_ERR_ICV_CHECK))
```

Comment:
Authentication error does not make much sense to me in outbound processing so I would rather have this always return alg error when !crypto.ok.

Reply:
Agreed. Do you want me to push as a separate patch or shall I make this change as part of stats implementation?

Reply:
Submitted a new PR as it is unrelated to stats implementation.

https://github.com/OpenDataPlane/odp/pull/1123